schematic/store/synapse.py

```diff
@@ -100,14 +101,32 @@ def __init__(
 
         self._query_fileview()
 
+    def _purge_synapse_cache(self):
+        # try clearing the cache
+        # scan a directory and check size of files
+        if os.path.exists("/var/www/.synapseCache"):
+            #nbytes = sum(d.stat().st_size for d in os.scandir('/var/www/.synapseCache') if d.is_file())
+            nbytes = get_dir_size("/var/www/.synapseCache")
+            # if 19.5 GB has already been taken, purge cache prior to today
+            if nbytes >= 20937965568:
+                c = self.syn.core.cache.Cache()
+                today = datetime.strftime(datetime.utcnow(), '%s')
+                c.purge(before_date = int(today))
+            else:
+                # print remaining ephemeral storage on AWS 
+                remaining_space = 21474836480 - nbytes
+                converted_space = convert_size(remaining_space)
+                print(f'Estimated {remaining_space} bytes (which is approximately {converted_space}) remained in ephemeral storage after calculating size of .synapseCache.')
+
     def _query_fileview(self):
+        self._purge_synapse_cache()
```

Comment:
@linglp the function makes sense, but, could this be a problem if you're running this function in parallel?  What happens when a query happens and then a purge occurs immediately after?

I'm not sure if as.DataFrame would work. I think it could be worthwhile to remove the folder size condition and test this function in a docker container (so you don't mess with your own cache).   Run this command in parallel many times to see if you run into an error.

Reply:
Thanks for pointing this out Tom. I tried using future concurrency to test out sending 20 concurrent requests. But maybe because clearing the cache took too long that I ended up getting: 
```
generated an exception:('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
generated an exception:('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
generated an exception:('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
generated an exception:('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
generated an exception:('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
generated an exception:('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
generated an exception:('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

```

Reply:
Note: problem about purging synapse cache before querying file view: two users could interrupt each other. For example, if user A download the file view, and user B purges the fileview before user A can read file view, then this would cause a problem for user A.